kedro/extras/datasets/video/video_dataset.py

```diff
@@ -0,0 +1,301 @@
+        if isinstance(index, slice):
+            return SlicedVideo(self, index)
+        return self.video[self.indexes[index]]
+
+    def __len__(self) -> int:
+        return len(self.indexes)
+
+    def __getattr__(self, item):
+        return getattr(self.video, item)
+
+
+class AbstractVideo(abc.Sequence):
+    """Base class for the underlying video data"""
+
+    _n_frames = 0
+    _index = 0  # Next available frame
+
+    @property
+    def fourcc(self) -> str:
+        """Get the codec fourcc specification"""
+        raise NotImplementedError()
+
+    @property
+    def fps(self) -> float:
+        """Get the video frame rate"""
+        raise NotImplementedError()
+
+    @property
+    def size(self) -> Tuple[int, int]:
+        """Get the resolution of the video"""
+        raise NotImplementedError()
+
+    def __len__(self) -> int:
+        return self._n_frames
+
+    def __getitem__(self, index: Union[int, slice]):
+        """Get a frame from the video"""
+        raise NotImplementedError()
+
+
+class FileVideo(AbstractVideo):
+    """A video object read from a file"""
+
+    def __init__(self, filepath: str) -> None:
+        self._filepath = filepath
+        self._cap = cv2.VideoCapture(filepath)
+        self._n_frames = self._get_length()
+
+    @property
+    def fourcc(self) -> str:
+        fourcc = self._cap.get(cv2.CAP_PROP_FOURCC)
+        return int(fourcc).to_bytes(4, "little").decode("ascii")
+
+    @property
+    def fps(self) -> float:
+        return self._cap.get(cv2.CAP_PROP_FPS)
+
+    @property
+    def size(self) -> Tuple[int, int]:
+        width = int(self._cap.get(cv2.CAP_PROP_FRAME_WIDTH))
+        height = int(self._cap.get(cv2.CAP_PROP_FRAME_HEIGHT))
+        return width, height
+
+    def __getitem__(self, index: Union[int, slice]):
+        if isinstance(index, slice):
+            return SlicedVideo(self, index)
+
+        if index < 0:
+            index += len(self)
+        if index >= len(self):
+            raise IndexError()
+
+        if index != self._index:
+            self._cap.set(cv2.CAP_PROP_POS_FRAMES, index)
+        self._index = index + 1  # Next frame to decode after this
+        ret, frame_bgr = self._cap.read()
+        if not ret:
+            raise IndexError()
+
+        height, width = frame_bgr.shape[:2]
+        return PIL.Image.frombuffer(  # Convert to PIL image with RGB instead of BGR
+            "RGB", (width, height), frame_bgr, "raw", "BGR", 0, 0
+        )
+
+    def _get_length(self) -> int:
+        # OpenCV's frame count might be an approximation depending on what
+        # headers are available in the video file
+        length = int(round(self._cap.get(cv2.CAP_PROP_FRAME_COUNT)))
+        if length >= 0:
+            return length
+
+        # On some video files it fails, we can count the frames but it will be
+        # very slow... It also seems like that opencv has issues to seek in those
+        # videos...
```

Comment:
Could you clean this comment a bit? Perhaps something like: 
```suggestion
        # Getting the frame count with OpenCV could fail on some video files, because
        # it would be very slow.
```